neighsyncd/neighsync.cpp

```diff
@@ -70,9 +75,18 @@ void NeighSync::onMsg(int nlmsg_type, struct nl_object *obj)
         return;
 
     key+= LinkCache::getInstance().ifindexToName(rtnl_neigh_get_ifindex(neigh));
+    intfName = key;
     key+= ":";
 
     nl_addr2str(rtnl_neigh_get_dst(neigh), ipStr, MAX_ADDR_SIZE);
+    /* Ignore IPv6 link-local addresses as neighbors, if ipv6 link local mode is disabled */
+    if (family == IPV6_NAME && IN6_IS_ADDR_LINKLOCAL(nl_addr_get_binary_addr(rtnl_neigh_get_dst(neigh))))
+    {
+        if (isLinkLocalEnabled(intfName) == false)
```

Comment:
This restricts IPv6 LLA nbr programming until the field ipv6_use_link_local_only is configured, why do we need this restriction? as long as we dont program the LLA nbr as host entry (but program it as NH entry) in the HW, we should be good, right?

I can have the global IPv6 address config on the interface and use LLA as NH for some routes without any ipv6_use_link_local_only config.

Reply:
Did we avoid programming Ipv6 LLA as neighbors into the HW?

Reply:
Yes right, we program LLA as nexthop only in the HW.
As @prsunny suggested to add IPv6 LLA neighbors only when "ipv6_use_link_local_only" is enabled on an interface.

Reply:
ok

Reply:
Agree with @venkatmahalingam, that LLA nexthop only must be programmed by orchagent and not LLA neighbors even though in this case, SAI is handling it. But this can be taken separately as the linklocal entries are now added to APPDB only in case of global config.